python/private/pypi/parse_requirements.bzl

```
@@ -315,26 +315,46 @@ def _add_dists(*, requirement, index_urls, logger = None):
     whls = []
     sdist = None
 
-    # TODO @aignas 2024-05-22: it is in theory possible to add all
-    # requirements by version instead of by sha256. This may be useful
-    # for some projects.
-    for sha256 in requirement.srcs.shas:
-        # For now if the artifact is marked as yanked we just ignore it.
-        #
-        # See https://packaging.python.org/en/latest/specifications/simple-repository-api/#adding-yank-support-to-the-simple-api
-
-        maybe_whl = index_urls.whls.get(sha256)
-        if maybe_whl and not maybe_whl.yanked:
-            whls.append(maybe_whl)
-            continue
-
-        maybe_sdist = index_urls.sdists.get(sha256)
-        if maybe_sdist and not maybe_sdist.yanked:
-            sdist = maybe_sdist
-            continue
-
-        if logger:
-            logger.warn(lambda: "Could not find a whl or an sdist with sha256={}".format(sha256))
+    # First try to find distributions by SHA256 if provided
+    if requirement.srcs.shas:
+        for sha256 in requirement.srcs.shas:
+            # For now if the artifact is marked as yanked we just ignore it.
+            #
+            # See https://packaging.python.org/en/latest/specifications/simple-repository-api/#adding-yank-support-to-the-simple-api
+
+            maybe_whl = index_urls.whls.get(sha256)
+            if maybe_whl and not maybe_whl.yanked:
+                whls.append(maybe_whl)
+                continue
+
+            maybe_sdist = index_urls.sdists.get(sha256)
+            if maybe_sdist and not maybe_sdist.yanked:
+                sdist = maybe_sdist
+                continue
+
+            if logger:
+                logger.warn(lambda: "Could not find a whl or an sdist with sha256={}".format(sha256))
+    else:
+        # If no SHA256s provided, try to find distributions by version
+        version = requirement.srcs.version
+        if version:
```

```suggestion
    elif requirement.srcs.version:
        # If no SHA256s provided, try to find distributions by version
```

could be a simple way to define it and it avoids having the indentation.